app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2141,4 +2143,32 @@ public void onClick(View view) {
     public enum Flag {
         RESET_DATABASE,
     }
+    public void setupNetworkBroadcastReceiver()
+    {
+        BroadcastReceiver networkStateReceiver = new BroadcastReceiver() {
+            final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.d("Network Listener", "Network Type Changed");
+                boolean isConnected = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
+                if(isConnected){
+                    //Disable fab when there is no connectivity
+                    fab.setEnabled(false);
```

Comment:
Its because we have a set a custom `app:layout_behaviour` in **main.xml** for our fab which does not include moving up on snackbar's appearance. The moving up on snackbar's appearance is included in the default behaviour of a fab though this can be fixed by creating a different custom layout behaviour class.